src/radical/pilot/utils/serializer.py

```diff
@@ -129,17 +129,25 @@ def deserialize_file(fname):
 #
 def deserialize_obj(obj):
     '''
-    Deserialize object from str
+    Deserialize object from str.
+
+    Raises
+    ------
+    TypeError
+        if object cannot be deserialized.
+    ValueError
+        if no result is found.
+
     '''
     result = None
 
     try:
         result = dill.loads(obj)
     except Exception as e:
-        raise Exception ("failed to deserialize from object") from e
+        raise TypeError("failed to deserialize from object") from e
 
     if result is None:
```

Comment:
@AymenFJA : can this actually happen, that `dill.loads` will not raise any exception but still returns no data?

Reply:
No, `dill loads/load` will always raise an error/exception if there is data corruption or unserialized object fed, etc... We do not need to reraise since the exception will fall normally.

Reply:
Right - then let's take this part out.

Reply:
I'm looking at this thread and I'm not sure @andre-merzky and @AymenFJA are talking about the same line. @andre-merzky  is asking whether it is necessary to check `if result is None`. I can't tell whether @AymenFJA is saying that `dill.loads` will not return `None`.

Can you please clarify whether you agree that the `if result is None` conditional should be removed?